src/averageAges.js

```diff
@@ -34,7 +50,28 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let sum = 0;
+  let counter = 0;
+  const mapOfPeople = people
+    .reduce((acc, person) => ({ ...acc, [person.mother]: person }), {});
+
+  withChildren === true
+    ? people.reduce((acc, person) => {
+      if (person.sex === 'f' && mapOfPeople[person.name] !== undefined) {
```

Comment:
in this case `mapOfPeople[person.name] !== undefined` -> `!!mapOfPeople[person.name]`, but ok